spec/controllers/api/v1/notes_controller_spec.rb

```diff
@@ -0,0 +1,254 @@
+require 'rails_helper'
+
+shared_examples 'fetching the expected notes' do
+  it 'responds with expected notes' do
+    expect(response_body.to_json).to eq(expected)
+  end
+
+  it 'responds with 200 status' do
+    expect(response).to have_http_status(:ok)
+  end
+end
+
+shared_examples 'creating with errors' do
+  it 'responds with expected error status' do
+    expect(response).to have_http_status(expected_status)
+  end
+
+  it 'responds with expected error message' do
+    expect(response_body['error']).to eq(expected_error_message)
+  end
+
+end
+describe Api::V1::NotesController, type: :controller do
+  describe 'GET #index' do
+    let(:expected) do
+      ActiveModel::Serializer::CollectionSerializer.new(notes_expected,
+                                                        serializer: IndexNoteSerializer).to_json
+    end
+    let(:params) { {} }
+    let(:utilities) { %i[north_utility south_utility] }
+    let(:create_user_notes) { create_list(:note, 5, user: user, utility: create(utilities.sample)) }
+
+    context 'when there is a user logged in' do
+      include_context 'with authenticated user'
+
+      context 'when fetching all the notes for user' do
+        let(:notes_expected) { create_user_notes }
+
+        before do
+          create_user_notes
+          get :index, params: params
+        end
+
+        it_behaves_like 'fetching the expected notes'
+      end
+
+      context 'when fetching notes with page and page size params' do
+        let(:page)            { 1 }
+        let(:page_size)       { 2 }
+        let(:notes_expected) { create_user_notes.first(2) }
+        let(:params) { { page: page, page_size: page_size } }
+
+        before do
+          create_user_notes
+          get :index, params: params
+        end
+
+        it_behaves_like 'fetching the expected notes'
+      end
+
+      context 'when fetching notes using filters' do
+        let(:create_user_reviews) { create_list(:note, 2, :review, user: user, utility: user.utility) }
```

Comment:
```suggestion
        let(:user_reviews) { create_list(:note, 2, :review, user: user, utility: user.utility) }
```
Cada vez que llamas al `user_reviews` no crea más, solamente memoiza lo ya creado.